Mage/src/main/java/mage/abilities/keyword/AwakenAbility.java

```diff
@@ -120,19 +120,21 @@ public AwakenEffect copy() {
         @Override
         public boolean apply(Game game, Ability source) {
             UUID targetId = null;
-            for (Target target : source.getTargets()) {
-                if (target.getFilter().getMessage().equals(filterMessage)) {
-                    targetId = target.getFirstTarget();
+            if (source != null && source.getTargets() != null) {

```

Comment:
I'm not sure if these ever can be null. What I do know is 100% of the card-related NPEs were with cards that have awaken on them (Clutch of Currents, Scatter to the Winds, Ruinous Path, Planar Outburst). So naturally I looked at those cards and the classes they use in common, AwakenAbility and AbilitiesImpl. Since these issues are very difficult to reproduce, I figured we might as well add some checks to prevent the many NPEs we are seeing from them. I'm open to better suggestions.